### PR TITLE
doc: Add code of conduct

### PR DESCRIPTION
**Pull request by kgriffs**
Add an official code of conduct, and reference it in the docs.

### CODE REVIEW

**Comment by swizzard** [comment]
:+1:

File: CODEOFCONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Code of Conduct
+
+All contributors and maintainers of this project are subject to this code of conduct.
+
+We pledge to respect everyone who contributes to the *Falcon* project or other associated activities by (including but not limited to) creating project issues, submitting pull requests, and providing feedback on the same. We also pledge to respect everyone who participates in discussions on the project's mailing list and IRC channel.
+
+Unacceptable behavior includes offensive verbal comments related to gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, ethnicity, religion, technology choices, sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention.
```
Comment (BenjamenMeyer) [inline CODEOFCONDUCT.md]: "includes but is not limited to" is a good phrase to use in these situations as you don't then have to list out each individual one, just the most common, or ones you want to specifically highlight.

Comment (kgriffs) [inline CODEOFCONDUCT.md]: Good suggestion; I'll add that.

Comment (kgriffs) [inline CODEOFCONDUCT.md]: Done.

**Comment by jmvrbanac** [comment]
:+1:

File: doc/community/contrib-snip.rst [hunk 2]
```diff
@@ -9,11 +9,12 @@ send an email to falcon@librelist.com and follow the instructions in the
 reply. For more information about managing your subscription, check out
 the `Librelist help page <http://librelist.com/help.html>`_.
 
-While we don't have an official code of conduct, we do expect everyone
-who participates on the mailing list to act professionally, and lead
-by example in encouraging constructive discussions. Each individual in
-the community is responsible for creating a positive, constructive, and
-productive culture.
+All contributors and maintainers of this project are subject to our `code
+of conduct <https://github.com/falconry/falcon/blob/master/CODEOFCONDUCT.md>`_.
```
Comment (fxfitz) [inline doc/community/contrib-snip.rst]: "Code of Conduct" (caps)

Its a thing, an important thing, so should probably be caps.

File: doc/community/contrib-snip.rst [hunk 3]
```diff
@@ -32,4 +33,9 @@ We'd like to invite you to help other community members with their
...
 `pull requests <https://github.com/racker/falcon/pulls>`_. If you use the
 Chrome browser, we recommend installing the
-`NotHub extension <http://nothub.org/>`_ to stay up to date with PRs.
\ No newline at end of file
+`NotHub extension <http://nothub.org/>`_ to stay up to date with PRs.
+
+Code of Conduct
+---------------
+All contributors and maintainers of this project are subject to our `code
+of conduct <https://github.com/falconry/falcon/blob/master/CODEOFCONDUCT.md>`_.
```
Comment (fxfitz) [inline doc/community/contrib-snip.rst]: See above.

File: doc/community/contrib-snip.rst (hunk 2 above)
Comment (kgriffs) [inline doc/community/contrib-snip.rst]: Done

File: doc/community/contrib-snip.rst (hunk 3 above)
Comment (kgriffs) [inline doc/community/contrib-snip.rst]: Done

**Comment by jmvrbanac** [comment]
:+1:

**Comment by fxfitz** [comment]
LGTM

**Comment by kgriffs** [comment]
[rebased] [edits commit title]

**Comment by jmvrbanac** [comment]
:+1:

**Pull request merged**